fe1-web/src/features/notification/screens/NotificationScreen.tsx

```diff
@@ -98,58 +51,132 @@ const NotificationScreen = () => {
-  const notificationData = useMemo(() => {
-    const addKey = (n: NotificationState): NotificationItem => ({
-      ...n,
-      key: n.id.toString(),
-      isLastItem: false,
-    });
-
-    const read: NotificationItem[] = unreadNotifications.map(addKey);
-    const unread: NotificationItem[] = readNotifications.map(addKey);
-
-    // these will be the items inside the FlatList component
-    const items: (ListSeparatorItem | NotificationItem)[] = [];
-
-    // add heading before the unread notifications if there are any
-    if (unread.length > 0) {
-      // set isLastItem property that allows having different styles
-      unread[unread.length - 1].isLastItem = true;
-
-      // the key property has to be unique. all notification keys are numbers so
-      // use a string starting with characters for separators
-      items.push(
-        { key: 'seperator:unread', title: 'Notifications' } as ListSeparatorItem,
-        ...unread,
-      );
-    }
+  return (
+    <ScreenWrapper>
+      <View style={List.container}>
+        <ListItem.Accordion
+          containerStyle={List.accordionItem}
+          content={
+            <ListItem.Content>
+              <ListItem.Title style={[Typography.base, Typography.important]}>
+                Notifications
+              </ListItem.Title>
+            </ListItem.Content>
+          }
+          onPress={() => setShowUnreadNotification(!showUnreadNotification)}
+          isExpanded={showUnreadNotification}>
+          {unreadNotifications.map((notification, idx) => {
+            const NotificationType = notificationTypes.find((t) => t.isOfType(notification));
+
+            if (!NotificationType) {
+              console.error('Unregistered notification type', notification);
+              throw new Error('Unregistered notification type');
+            }
+
+            const listStyle = List.getListItemStyles(
+              idx === 0,
+              idx === unreadNotifications.length - 1,
+            );
+
+            return (
+              <ListItem
+                key={notification.id}
+                containerStyle={listStyle}
+                style={listStyle}
+                onPress={() =>
+                  navigation.navigate<'Notification'>(
+                    STRINGS.navigation_notification_single_notification,
+                    {
+                      notificationId: notification.id,
+                    },
+                  )
+                }>
+                <View style={List.icon}>
+                  <NotificationType.Icon size={Icon.size} color={Color.primary} />
+                </View>
+                <ListItem.Content>
+                  <ListItem.Title style={Typography.base}>{notification.title}</ListItem.Title>
+                  <ListItem.Subtitle style={Typography.small}>
+                    <ReactTimeago date={notification.timestamp * 1000} />
+                  </ListItem.Subtitle>
+                </ListItem.Content>
+              </ListItem>
```

Comment:
I feel like it is very similar for both notification types. It could be nice to create a function to remove this duplication if it doesn't introduce too many arguments.

Reply:
I'll try it :)